src/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Windows.cs

```diff
@@ -410,61 +410,66 @@ private static ModuleInfo[] GetModuleInfos(int processId, bool firstModuleOnly)
-                            && moduleInfo._fileName.Length >= 4
-                            && moduleInfo._fileName.StartsWith(@"\\?\", StringComparison.Ordinal))
+                        // If the user is only interested in the main module, break now.
+                        // This avoid some waste of time. In addition, if the application unloads a DLL
+                        // we will not get an exception. 
+                        if (firstModuleOnly)
                         {
-                            moduleInfo._fileName = moduleInfo._fileName.Substring(4);
+                            break;
                         }
 
-                        moduleInfos.Add(moduleInfo);
+                        baseName.Clear();
+                        fileName.Clear();
                     }
-                    catch (Win32Exception e)
+
+                    IntPtr moduleHandle = moduleHandles[i];
+                    Interop.mincore.NtModuleInfo ntModuleInfo = new Interop.mincore.NtModuleInfo();
+                    if (!Interop.mincore.GetModuleInformation(processHandle, moduleHandle, ntModuleInfo, Marshal.SizeOf(ntModuleInfo)))
                     {
-                        if (e.NativeErrorCode == Interop.mincore.Errors.ERROR_INVALID_HANDLE || e.NativeErrorCode == Interop.mincore.Errors.ERROR_PARTIAL_COPY)
-                        {
-                            // It's possible that another thread casued this module to become
-                            // unloaded (e.g FreeLibrary was called on the module).  Ignore it and
-                            // move on.
-                        }
-                        else
-                        {
-                            throw;
-                        }
+                        HandleError();
+                        continue;
                     }
 
-                    //
-                    // If the user is only interested in the main module, break now.
-                    // This avoid some waste of time. In addition, if the application unloads a DLL                     
-                    // we will not get an exception. 
-                    //
-                    if (firstModuleOnly) { break; }
+                    ModuleInfo moduleInfo = new ModuleInfo
+                    {
+                        _sizeOfImage = ntModuleInfo.SizeOfImage,
+                        _entryPoint = ntModuleInfo.EntryPoint,
+                        _baseOfDll = ntModuleInfo.BaseOfDll
+                    };
+
+                    int ret = Interop.mincore.GetModuleBaseName(processHandle, moduleHandle, baseName, baseName.Capacity * 2);
+                    if (ret == 0)
+                    {
+                        HandleError();
+                        continue;
+                    }
+
+                    moduleInfo._baseName = baseName.ToString();
+                    ret = Interop.mincore.GetModuleFileNameEx(processHandle, moduleHandle, fileName, fileName.Capacity * 2);
```

Comment:
Nit: I'd add a blank line after setting moduleInfo._baseName.  Otherwise the grouping makes it look like we're setting it in order to prepare for the GetModuleFileNameEx call.